lib/utils/export/to-zip.js

```diff
@@ -0,0 +1,99 @@
+import JSZip from 'jszip';
+import sanitize from 'sanitize-filename';
+import { identity, update } from 'lodash';
+
+/**
+ * Generates filename for note based on note content
+ *
+ * Please see code for algorithm. The goal is to
+ * find the first usable and non-blank line of text
+ * from the note to generate the title.
+ *
+ * @param {Object} note object
+ * @returns {Object} augmented note object with new filename
+ */
+const addFilename = note => ( {
+	...note,
+	fileName: note.content
+		.split( '\n' ) // base filename off of a single line
+		.map( line => line.trim() ) // and ignore leading/trailing spaces
+		.map( sanitize ) // strip away any invalid characters for a filename (such as `/`)
+		.filter( identity ) // remove blank lines
+		.concat( 'untitled' ) // use this as a default if there are no non-blank lines
+		.shift() // take the first remaining line
+		.slice( 0, 40 ), // and truncate to the first 40 characters
+} );
+
+/**
+ * Appends associated tags as a list at end of note content if available
+ *
+ * @example
+ * // returns the following
+ * """
+ * a regular plumbus
+ *
+ * Tags:
+ *  - #plumbus
+ *  - #howisitmade
+ * """
+ * appendTags( { content: 'a regular plumbus', tags: [ 'plumbus', 'howisitmade' ] } )
+ *
+ * @example
+ * // returns note content unchanged
+ * appendTags( { content: 'the fleeb has the fleeb juice' } )
+ *
+ * @param {Object} note note object
+ * @returns {Object} augmented note whose
+ */
+const appendTags = note => note.tags
+	? { ...note, content: `${ note.content }\n\nTags:\n${ note.tags.map( tag => ` - #${ tag }` ).join( '\n' ) }` }
```

Comment:
If we are using hashtags, I would prefer that they are just listed one each per line with out the ` - `:

```
Tags:
#todo
#shopping
```

Thoughts? I'm actually backtracking a bit now on using the hash because we don't use it anywhere else in the apps or website.